pkg/flowaggregator/flowaggregator.go

```diff
@@ -194,7 +275,28 @@ func (fa *flowAggregator) Run(stopCh <-chan struct{}) {
 					continue
 				}
 			}
-			err := fa.aggregationProcess.ForAllRecordsDo(fa.sendFlowKeyRecords)
+			dataSet := ipfix.NewSet(ipfixentities.Data, fa.templateID, false)
+			addFlowKeyRecord := func(key ipfixintermediate.FlowKey, record ipfixintermediate.AggregationFlowRecord) error {
+				if !record.ReadyToSend {
+					klog.V(4).Info("Skip sending record that is not correlated.")
+					return nil
+				}
+				// limit dataSet to have no more than two records
+				// TODO: more records per dataSet will be supported when go-ipfix supports size check when adding records
+				if dataSet.GetSet().GetNumberOfRecords() == recordNumPerSet {
+					_, err := fa.sendDataSet(dataSet)
+					if err != nil {
+						return err
+					}
+					dataSet = ipfix.NewSet(ipfixentities.Data, fa.templateID, false)
+				}
+				err := dataSet.AddRecord(record.Record.GetOrderedElementList(), fa.templateID)
+				if err != nil {
+					return fmt.Errorf("AddRecord failed, error: %v", err)
```

Comment:
not a very good error message - I don't think you should refer to the function name but instead explain in plain english what failed